packages/plugin-commands-installation/src/update/index.ts

```diff
@@ -234,6 +235,8 @@ async function update (
     allowNew: false,
     includeDirect,
     update: true,
+    updateMatching: dependencies.every(dep => !dep.substr(1).includes('@')) && opts.depth && opts.depth > 0 && !opts.latest
```

Comment:
FYI, the Mozilla docs have this to say about `substr`:

Warning: Although String.prototype.substr(…) is not strictly deprecated (as in "removed from the Web standards"), it is considered a legacy function and should be avoided when possible. It is not part of the core JavaScript language and may be removed in the future. If at all possible, use the substring() method instead.